src/sync.c

```diff
@@ -220,23 +222,41 @@ void sync_process(sync_t *st)
         }
         else if (st->cfo_wait == 0)
         {
-            for (i = -2 * PARTITION_WIDTH; i < 2 * PARTITION_WIDTH; ++i)
+            for (cfo = -2 * PARTITION_WIDTH; cfo < 2 * PARTITION_WIDTH; cfo++)
```

Comment:
I wouldn't be opposed to pulling this code out to a separate function if you want. It looks independent of the other code.